src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs

```diff
@@ -209,5 +207,33 @@ static void RegisterMessageRoute(Type mappedType, string address, Dictionary<Typ
             }
             routeTableEntries[mappedType] = new RouteTableEntry(mappedType, route);
         }
+
+        static List<Type> GetBaseTypes(Type messageType, Conventions conventions)
+        {
+            var result = new List<Type>();
+            var baseType = messageType.BaseType;
```

Comment:
@andreasohlund see https://github.com/Particular/NServiceBus/pull/4097/files/82142c44bd97b6a6516123c17619ed8746194552#diff-d9f6e42c7a4ae0af9eb4c474ce658281R228 I get the implemented interfaces. Isn't that enough? At least serializer does it that way as well